### PR TITLE
rocAL - Python Unit test fix

### PR DESCRIPTION
**Pull request by swetha097**
This PR fixes issues with Python unit test

### CODE REVIEW

**Comment by paveltc** [comment]
I see the following error when running with GPU:
[ERR]  {rocalVerify}   { verify } vxVerifyGraph failed -1

File: rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py [hunk 1]
```diff
@@ -103,6 +105,8 @@ def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = Fals
                     self.out = torch.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=torch.float32, device = torch_gpu_device)
                 elif self.tensor_dtype == types.FLOAT16:
                     self.out = torch.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=torch.float16, device = torch_gpu_device)
+                elif self.tensor_dtype ==types.UINT8:
+                    self.out = torch.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=torch.uint8, device = torch_gpu_device)
                 torch_gpu_device = torch.device('cuda', self.device_id)
```
Comment (rrawther) [inline rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py:110]: looks like torch_gpu_device is defines again in this line. Please remove duplicate definitions (applies to other such cases)

File: rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py [hunk 2]
```diff
@@ -120,6 +126,8 @@ def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = Fals
                     self.out = torch.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=torch.float32, device=torch_gpu_device)
                 elif self.tensor_dtype == types.FLOAT16:
                     self.out = torch.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=torch.float16, device=torch_gpu_device)
+                elif self.tensor_dtype == types.UINT8:
+                    self.out = torch.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=torch.uint8, device=torch_gpu_device)
                 torch_gpu_device = torch.device('cuda', self.device_id)
```
Comment (rrawther) [inline rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py:131]: remove duplicate definition

**Review by rrawther — CHANGES_REQUESTED** [review]

**Comment by LakshmiKumar23** [comment]
@swetha097 @paveltc I do not see the verifyGraph error, but it doesn't run any of the augmentations for gpu backend. The output folders are empty. I also only this as printed output on terminal. Seems like only one hot runs with GPU backend.
```
###############################################                             ONE_HOT                         ############################################
###############################################                             SUCCESS                             ###############################################
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
OK: loaded 82 kernels from libvx_rpp.so
```

**Comment by swetha097** [comment]
> @swetha097 @paveltc I do not see the verifyGraph error, but it doesn't run any of the augmentations for gpu backend. The output folders are empty. I also only this as printed output on terminal. Seems like only one hot runs with GPU backend.
> 
> ```
> ###############################################                             ONE_HOT                         ############################################
> ###############################################                             SUCCESS                             ###############################################
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> OK: OpenVX using GPU device - 0: AMD Radeon Graphics [gfx1030] with 80 CUs on PCI bus 0d:00.0
> OK: loaded 82 kernels from libvx_rpp.so
> ```

@LakshmiKumar23 I am not seeing this issue in ixt-sjc2-54 and MI200 Machine where I tested. All augmentations run to completion with the success statement for GPU backend.

@paveltc I do not see the verify graph failure for GPU backend.

The issues I am seeing are some discrepancies for Vignette, RMN, CMN, Nop & Copy in gpu backend.

**Comment by LakshmiKumar23** [comment]
@swetha097 Finally got it running. CPU fails on copy
image

Can you also sync with TOT. I think @rrawther cleaned the code to get rid of some print statements. I still see them on my end. Like "in while loop"

For GPU: it does not take the argument. THat's if I call with ```sh PYTHON_UNITTEST_TEST_FILE.sh -b gpu```.
If I give ```./PYTHON_UNITTEST_TEST_FILE -b gpu```, it takes the gpu arg but fails to run all augmentations
image

File: rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py (hunk 1 above)
Comment (swetha097) [reply rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py]: done

File: rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py (hunk 2 above)
Comment (swetha097) [reply rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py]: done

**Comment by swetha097** [comment]
@LakshmiKumar23  Merged with the TOT develop branch. Can see that the copy works fine for both cpu & gpu backend.

But after the merge, I am seeing the file changes that are not from my PR too.

**Comment by paveltc** [comment]
@swetha097 @LakshmiKumar23 @rrawther I just ran both CPU and GPU, it seems to run fine with no errors.
Can the output log file please be changed to have a name that is easy to copy to the terminal?
Currently it is run.rocAL_api_log.YYYY-MM-DD-HH:MM:SS
Can this be changed so that there are not : in the time stamp? And can the log be modified to have a summary of which tests ran and passed?
Thanks,
Pavel

**Comment by swetha097** [comment]
@LakshmiKumar23 @rrawther @paveltc 

I am closing this PR and creating a new one as this shows changes from other merged PR's too.

@paveltc  We will create a new PR for the suggested changes.

**Pull request closed without merging**